File: exercises/day13-Keyboard-First-Refactoring/typescript/src/elf/Elf.ts
import { SleightTrip } from '../trip/SleightTrip';

/**
 * Represents an elf in Santa's workshop.
 *
 * Each elf:
 *  - has a list of workshop friends (other elves)
 *  - has a list of sleigh trips they went on to deliver presents
 */
export class Elf {
    private sleighTrips: SleightTrip[] = [];
    private friends: Elf[] = [];

    getFriends(): Elf[] {
        return this.friends;
    }

    addFriend(elf: Elf): void {
        this.friends.push(elf);
    }

    addTrip(sleighTrip: SleightTrip): void {
        this.sleighTrips.push(sleighTrip);
    }

    trips(): SleightTrip[] {
        return this.sleighTrips;
    }
}

File: exercises/day13-Keyboard-First-Refactoring/java/src/main/java/sleightrip/elf/Elf.java
package sleightrip.elf;

import java.util.ArrayList;
import java.util.List;

import sleightrip.trip.SleightTrip;

/**
 * Represents an elf in Santa's workshop.
 *
 * Each elf:
 *  - has a list of workshop friends (other elves)
 *  - has a list of sleigh trips they went on to deliver presents
 */
public class Elf {

    private List<SleightTrip> sleighTrips = new ArrayList<>();
    private List<Elf> friends = new ArrayList<>();

    public List<Elf> getFriends() {
        return friends;
    }

    public void addFriend(Elf elf) {
        friends.add(elf);
    }

    public void addTrip(SleightTrip sleighTrip) {
        sleighTrips.add(sleighTrip);
    }

    public List<SleightTrip> trips() {
        return sleighTrips;
    }
}
